packages/ember-views/lib/components/component.js

```diff
@@ -132,12 +132,33 @@ var Component = View.extend(TargetActionSupport, {
   }),
 
   init() {
-    this._super.apply(this, arguments);
+    this._super(...arguments);
     set(this, 'controller', this);
     set(this, 'context', this);
+
+    if (this.layoutName && this.container) {
+      let layoutName = get(this, 'layoutName');
+
+      set(this, 'layout', this.templateForName(layoutName));
```

Comment:
To answer the actual question though: we only check for them once after initial creation, and they are not observed for changes.